packages/typescript-checker/test/unit/grouping/create-groups.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { expect } from 'chai';
+
+import { factory } from '@stryker-mutator/test-helpers';
+
+import { Node } from '../../../src/grouping/node.js';
+
+import { createGroups } from '../../../src/grouping/create-groups.js';
+
+describe('create-group createGroups', () => {
+  it('single mutant should create single group', () => {
+    const mutants = [factory.mutant({ fileName: 'a.js', id: '1' })];
+    const mutantsClone = structuredClone(mutants);
+    const nodes = new Map<string, Node>([['a.js', new Node('a.js', [], [])]]);
+    const groups = createGroups(mutants, nodes);
+    expect(groups).to.have.lengthOf(1);
+    expect(groups[0][0]).to.be.equal(mutantsClone[0].id);
+  });
```

Comment:
```suggestion
  it('single mutant should create single group', () => {
    const mutants = [factory.mutant({ fileName: 'a.js', id: 'mutant-1' })];
    const nodes = new Map<string, Node>([['a.js', new Node('a.js', [], [])]]);
    const groups = createGroups(mutants, nodes);
    expect(groups).to.have.lengthOf(1);
    expect(groups[0]).to.have.lengthOf(1);
    expect(groups[0][0]).to.be.equal('mutant-1');
  });
```

`structuredClone` [just landed in node 18](https://github.com/nodejs/node/pull/39759). Since we still support node 14, we cannot use it.

Since we're only using it to keep the order of mutant arrays and pick off the `id` fields which we've hard-coded in the first place, i think we should use those hard-coded ids directly.